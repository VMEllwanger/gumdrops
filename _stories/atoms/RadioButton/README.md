## Props

The following props may be passed to configure the Radio Button. All additional props will be applied to the `<input>`

| name      | type       | required | description                                                                          |
| --------- | ---------- | -------- | ------------------------------------------------------------------------------------ |
| className | `String`   |          | Additional class to be added to the outermost element.                               |
| context   | `String`   |          | Indicate the context of the button. Example `success`, `warning`, `danger` etc.      |
| label     | `String`   |          | The label text to apply to the radio button.                                         |
| value     | `String`   |          | The value to apply to the radio button.                                              |
| name      | `String`   | true     | Identifier that will be set on individual radio inputs.                              |
| onBlur    | `Function` |          | Blur handler that will be called when a blur event happens on a `<RadioButton>`.     |
| onChange  | `Function` |          | Change handler that will be called when a change event happens on a `<RadioButton>`. |
