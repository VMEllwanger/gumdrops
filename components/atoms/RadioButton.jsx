import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

const RadioButton = ({ value, label, onChange, onBlur, name, context, className, ...rest }) => {
    const baseClass = 'gds-form-group';

    const rootClass = cx(baseClass, `${baseClass}__radio-label`, className, {
        [`${baseClass}--${context}`]: context
    });

    return (
        <label className={rootClass}>
            <input
                className="gds-form-group__radio-input"
                type="radio"
                name={name}
                onChange={onChange}
                onBlur={onBlur}
                value={value}
                {...rest}
            />
            <span className="gds-form-group__radio-indicator" />
            {label}
        </label>
    );
};

RadioButton.propTypes = {
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]).isRequired,
    name: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    context: PropTypes.string,
    className: PropTypes.string,
    /** xs, sm */
    onChange: PropTypes.func,
    onBlur: PropTypes.func
};

export default RadioButton;
