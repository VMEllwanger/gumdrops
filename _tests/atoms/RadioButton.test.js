/* globals shallow */
import React from 'react';
import RadioButton from '../../components/atoms/RadioButton';

const defaultProps = {
    label: 'Cool',
    value: 'cool',
    className: 'custom-class',
    context: 'success',
    name: 'radio-buttons',
    onBlur: () => {},
    onChange: () => {}
};

describe('Expect <RadioButton>', () => {
    it('to render', () => {
        const wrapper = shallow(<RadioButton {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
