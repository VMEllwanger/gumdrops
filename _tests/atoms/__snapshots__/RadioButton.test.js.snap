// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <RadioButton> to render 1`] = `
<label
  className="gds-form-group gds-form-group__radio-label custom-class gds-form-group--success"
>
  <input
    className="gds-form-group__radio-input"
    name="radio-buttons"
    onBlur={[Function]}
    onChange={[Function]}
    size="sm"
    type="radio"
    value="cool"
  />
  <span
    className="gds-form-group__radio-indicator"
  />
  Cool
</label>
`;
